assets/js/modules/adsense/dashboard/adsense-module-status.js

```diff
@@ -24,131 +24,304 @@ import SvgIcon from 'GoogleUtil/svg-icon';
-		this.setState( { ...this.state, loadStatus } );
 	}
 
 	async componentDidMount() {
-		const existingTag = await getExistingTag( 'adsense' );
+		await this.updateAccountStatus();
+	}
 
-		this.setState( { existingTag } );
+	componentDidUpdate( previousProps, previousState ) {
+		const { accountStatus } = this.state;
 
-		getAdSenseAccountStatus( this.updateLoadStatus, existingTag ).then( ( results ) => {
-			this.setState( results );
-		} );
+		if ( previousState.accountStatus !== accountStatus ) {
+			const adSenseSetupInstructionsProps = propsFromAccountStatus( accountStatus );
+			this.setState( { instructionProps: adSenseSetupInstructionsProps } );
+		}
 	}
 
 	/**
 	 * If setup requires a continue step, the method repopulates state with the new data.
+	 * @param {Object} continueData
 	 */
 	continueSetup( continueData ) {
-		continueData.existingState = this.state;
-		this.setState( continueData );
+		const { instructionProps } = this.state;
+
+		continueData.existingState = { ...instructionProps };
+		this.setState( { instructionProps: continueData } );
 	}
 
 	/**
 	 * Go back to the previous (existing) state.
 	 */
 	goBack() {
-		const { existingState } = this.state;
+		const { instructionProps } = this.state;
+
+		const { existingState } = { ...instructionProps };
 		if ( existingState ) {
 			existingState.existingState = false;
-			this.setState( existingState );
+			this.setState( { instructionProps: existingState } );
 		}
 	}
 
-	handleComponentRender() {
-		const {
-			accountStatus,
-			loadStatus,
-		} = this.state;
-
-		if ( ! googlesitekit.canAdsRun && ! googlesitekit.modules.adsense.setupComplete ) {
-			return (
-				<Fragment>
-					<div className="googlesitekit-settings-module-warning">
-						<SvgIcon id="error" height="20" width="23" /> { __( 'Ad blocker detected, you need to disable it in order to setup AdSense.', 'google-site-kit' ) }
-					</div>
-				</Fragment>
-			);
-		}
-
-		if ( ! accountStatus ) {
-			return (
-				<Fragment>
-					{ loadStatus }
-					<ProgressBar />
-				</Fragment>
-			);
-		}
-		if ( 'account-pending-review' === accountStatus ) {
-			return (
-				<AdSenseInProcessStatus status="incomplete" />
-			);
-		}
-
-		if ( 'account-required-action' === accountStatus ) {
-			return (
-				<AdSenseInProcessStatus status="requiredAction" />
-			);
-		}
+	async updateAccountStatus() {
+		const existingTag = await getExistingTag( 'adsense' );
+		const setLoadingMessage = ( message ) => {
+			this.setState( { loadingMessage: message } );
+		};
 
-		if ( 'ads-display-pending' === accountStatus ) {
-			return (
-				<AdSenseInProcessStatus status="adsDisplayPending" />
-			);
-		}
+		const status = await getAdSenseAccountStatus( existingTag, setLoadingMessage );
 
-		return (
-			<AdSenseSetupInstructions
-				{ ...this.state }
-				continueSetup={ this.continueSetup }
-				goBack={ this.goBack }
-			/>
-		);
+		this.setState( { accountStatus: status.accountStatus } );
```

Comment:
We don't use the `clientID` returned from the function at all - that seems problematic. Or otherwise, why is it returned at all then?

Reply:
The previous functions were returning a _lot_ of stuff without much documentation. I removed a lot but missed this one; it's since been removed. 👍